crates/bevy_color/src/hsla.rs

```diff
@@ -166,6 +168,35 @@ impl Luminance for Hsla {
     }
 }
 
+impl ClampColor for Hsla {
+    fn clamped(&self) -> Self {
+        Self {
+            hue: self.hue.rem_euclid(360.),
+            saturation: self.saturation.clamp(0., 1.),
+            lightness: self.lightness.clamp(0., 1.),
+            alpha: self.alpha.clamp(0., 1.),
+        }
+    }
+
+    fn clamp(&mut self) {
+        self.hue = self.hue.rem_euclid(360.);
+        self.saturation = self.saturation.clamp(0., 1.);
+        self.lightness = self.lightness.clamp(0., 1.);
+        self.alpha = self.alpha.clamp(0., 1.);
+    }
+
+    fn is_within_bounds(&self) -> bool {
+        self.hue >= 0.
```

Comment:
I think it's more idiomatic in Rust to say `(0. ..=360.).contains(&self.hue)`. At least, clippy prints a warning for this.